images/linux/scripts/installers/pypy.sh

```diff
@@ -69,28 +69,11 @@ function InstallPyPy
     rm -f $PACKAGE_TAR_TEMP_PATH
 }
 
-function getPyPyVersions
-{
-    uri="https://downloads.python.org/pypy/"
-    i=20
-
-    while [ $i -gt 0 ]; do
-        ((i--))
-        result="$(curl -4 -s --compressed $uri | grep 'linux64' | awk -v uri="$uri" -F'>|<' '{print uri$5}')"
-        if [ -z "$result" ]; then
-            sleep 30
-        else
-            echo "$result"
-            return
-        fi
-    done
-
-    echo "There are no more attempts to retrive PyPy version"
-    exit 1
-}
-
 # Installation PyPy
-pypyVersions=$(getPyPyVersions)
+uri="https://downloads.python.org/pypy/"
+download_with_retries $uri "." "pypyUrls.html"
+pypyVersions="$(cat pypyUrls.html | grep 'linux64' | awk -v uri="$uri" -F'>|<' '{print uri$5}')"

```

Comment:
Do we want the separate retry function for curl?
Looks like file is not useful here?